### ISSUE TITLE
Cannot save profile background on public profile

### ISSUE THREAD

**Issue by xXbwarsXx** [rhunk/SnapEnhance#859]
### Description

So I read the changelog and browsed through the new UI and saw this "Download profile picture" enabled it, tried it out and pressing on Download background comes up with this (see attached picture)
![Screenshot_20240410_085619_Snapchat](https://github.com/rhunk/SnapEnhance/assets/90275114/47a58f53-7219-4306-828c-f7c0367bd539)
![Screenshot_20240410_090804_SnapEnhance](https://github.com/rhunk/SnapEnhance/assets/90275114/111347fb-40ad-4df0-ab4c-ddffb68afd8e)

### Reproduction steps

1. Enable the feature to download profile pictures
2. Go to a friends public profile
3. Click on "Download profile picture"
4. Select "Background"
5. See error.

### Snapchat Version

12.80.1.0

### SnapEnhance Version

2.0

### Agreement

- [X] This is not a bug regarding Snapchat+.
- [X] I have provided a detailed description of the issue.
- [X] I have attached a log if deemed neccessary.
- [X] This issue is not a duplicate.

**Comment by bocajthomas** [comment]
Is it a video or photo ?

**Comment by bocajthomas** [comment]
I have Tested This On Snapchat V12.79 & On V12.81 ( Working ) 

- Make Sure You Have Force Image Format To Ether jpg or png
- Make Sure You Have set a download folder for snapEnhance ( this isn't really necessary since it will fall back to the folder you selected on setup page ) 
-  And then toggle On " Download Profile Photo " 



> * [x]  I have attached a log if deemed necessary.

You have Ticked That you have provided logs which You haven't done 
Please Provide Me Logs!! 

<details closed>
  <summary>How to do this</summary>

1. Open SnapEnhance 
2. On Home Tab 
3. Tap the "Bug" icon 
4. Tap The 3 Dots 
5. Press "Export Logs"
6. Then Send on this Issue 

</details>

**Comment by xXbwarsXx** [comment]
> I have Tested This On Snapchat V12.79 & On V12.81 ( Working )
> 
> * Make Sure You Have Force Image Format To Ether jpg or png
> * Make Sure You Have set a download folder for snapEnhance ( this isn't really necessary since it will fall back to the folder you selected on setup page )
> * And then toggle On " Download Profile Photo "
> 
> > * [x]   I have attached a log if deemed necessary.
> 
> You have Ticked That you have provided logs which You haven't done Please Provide Me Logs!!
> 
> How to do this

So sorry, absolutely misunderstood that checkbox. Original comment has been updated to include logs now. Hope that is enough.

**Comment by bocajthomas** [comment]
> Also, I forced the image format to jpg, set a snapenhance download folder and it still fails.

Try updating Snapchat to V12.81 from [Google play Store](https://play.google.com/store/apps/details?id=com.snapchat.android&pcampaignid=web_share)

**Comment by xXbwarsXx** [comment]
> > Also, I forced the image format to jpg, set a snapenhance download folder and it still fails.
> 
> Try updating Snapchat to V12.81 from [Google play Store](https://play.google.com/store/apps/details?id=com.snapchat.android&pcampaignid=web_share)

Same problem with 12.81.0.44 the latest one from the play store

**Comment by bocajthomas** [comment]
> Same problem with 12.81.0.44 the latest one from the play store

It seams to do it with some user's profile pic 

some work and some Dont 
@rhunk possible fix ?

**Comment by xXbwarsXx** [comment]
> > Same problem with 12.81.0.44 the latest one from the play store
> 
> It seams to do it with some user's profile pic
> 
> some work and some Dont @rhunk possible fix ?

Tried it with Phil Walton's profile (lens creator) still nothing, just gives me an error

**Comment by bocajthomas** [comment]
> Tried it with Phil Walton's profile (lens creator) still nothing, just gives me an error

I Tried With ( [Wow Clean](https://story.snapchat.com/p/5e7d315a-f401-4f78-b78e-30dc0c06ef49)
It worked

**Comment by xXbwarsXx** [comment]
> > Tried it with Phil Walton's profile (lens creator) still nothing, just gives me an error
> 
> I Tried With ( [Wow Clean](https://story.snapchat.com/p/5e7d315a-f401-4f78-b78e-30dc0c06ef49) It worked

How weird, "Wow Clean" worked for me too. So, the problem still exists but only at some accounts. Interesting.

**Comment by bocajthomas** [comment]
> How weird, "Wow Clean" worked for me too. So, the problem still exists but only at some accounts. Interesting.

Yh it should be fixed in the future hopefully

**Comment by xXbwarsXx** [comment]
> Fixed in [79a58ee](https://github.com/rhunk/SnapEnhance/commit/79a58ee89e84ba8d06c5b0e7e0e0022295c561a2)

And umm, what am I supposed to do with this? 😅
I'm kinda lost, will this be introduced in an update or can I fix it myself somehow?

**Comment by authorisation** [comment]
Yes this issue is fixed in commit https://github.com/rhunk/SnapEnhance/commit/79a58ee89e84ba8d06c5b0e7e0e0022295c561a2 and you can use it whenever a new build releases.

There is also the option to compile it yourself, you pick.
